utils/results/common.go

```diff
@@ -0,0 +1,612 @@
+		}
+	}
+	return getFinalApplicabilityStatus(applicableStatuses)
+}
+
+func getApplicabilityStatusFromRule(rule *sarif.ReportingDescriptor) jasutils.ApplicabilityStatus {
+	if rule.Properties[jasutils.ApplicabilitySarifPropertyKey] != nil {
+		status, ok := rule.Properties[jasutils.ApplicabilitySarifPropertyKey].(string)
+		if !ok {
+			log.Debug(fmt.Sprintf("Failed to get applicability status from rule properties for rule_id %s", rule.ID))
+		}
+		switch status {
+		case "not_covered":
+			return jasutils.NotCovered
+		case "undetermined":
+			return jasutils.ApplicabilityUndetermined
+		case "not_applicable":
+			return jasutils.NotApplicable
+		case "applicable":
+			return jasutils.Applicable
+		}
+	}
+	return ""
+}
+
+func GetDependencyId(depName, version string) string {
+	return fmt.Sprintf("%s:%s", depName, version)
+}
+
+func GetScaIssueId(depName, version, issueId string) string {
+	return fmt.Sprintf("%s_%s_%s", issueId, depName, version)
+}
+
+// GetUniqueKey returns a unique string key of format "vulnerableDependency:vulnerableVersion:xrayID:fixVersionExist"
+func GetUniqueKey(vulnerableDependency, vulnerableVersion, xrayID string, fixVersionExist bool) string {
+	return strings.Join([]string{vulnerableDependency, vulnerableVersion, xrayID, strconv.FormatBool(fixVersionExist)}, ":")
+}
+
+// Relevant only when "third-party-contextual-analysis" flag is on,
+// which mean we scan the environment folders as well (node_modules for example...)
+// When a certain package is reported applicable, and the evidence found
+// is inside the source code of the same package, we should disqualify it.
+//
+// For example,
+// Cve applicability was found inside the 'mquery' package.
+// filePath = myProject/node_modules/mquery/badCode.js , disqualify = True.
+// Disqualify the above evidence, as the reported applicability is used inside its own package.
+//
+// filePath = myProject/node_modules/mpath/badCode.js  , disqualify = False.
+// Found use of a badCode inside the node_modules from a different package, report applicable.
+func shouldDisqualifyEvidence(components map[string]services.Component, evidenceFilePath string) (disqualify bool) {
+	for key := range components {
+		if !strings.HasPrefix(key, techutils.Npm.GetPackageTypeId()) {
+			return
+		}
+		dependencyName := extractDependencyNameFromComponent(key, techutils.Npm.GetPackageTypeId())
+		// Check both Unix & Windows paths.
+		if strings.Contains(evidenceFilePath, nodeModules+"/"+dependencyName) || strings.Contains(evidenceFilePath, filepath.Join(nodeModules, dependencyName)) {
+			return true
+		}
+	}
+	return
+}
+
+func extractDependencyNameFromComponent(key string, techIdentifier string) (dependencyName string) {
+	packageAndVersion := strings.TrimPrefix(key, techIdentifier)
+	split := strings.Split(packageAndVersion, ":")
+	if len(split) < 2 {
+		return
+	}
+	dependencyName = split[0]
```

Comment:
I dont know if it applied here and how the dependencies are getting here but in some package managers, like Gradle, we can get a dependency like this: 'junit:junit:4.7'
the split will take the first 'junit' after splitting by :, although this is not the dependency's name, but the dependency group (in this case they are similar, but it is not always the case).